src/frontend/src/utils/stream_graph_formatter.rs

```diff
@@ -93,6 +135,89 @@ impl StreamGraphFormatter {
         writeln!(f, "{}{}", " ".repeat(level * 2), one_line_explain)?;
+        let explain_table_oneline = match node.get_node_body().unwrap() {
+            stream_node::NodeBody::Materialize(node) => Some(format!(
+                "materialized table: {}",
+                self.add_table(node.get_table().unwrap())
+            )),
+            stream_node::NodeBody::GlobalSimpleAgg(node) => Some(format!(
+                "state tables: [{}]",
+                node.internal_tables
+                    .iter()
+                    .map(|tb| { self.add_table(tb) })
+                    .join(", ")
+            )),
+            stream_node::NodeBody::HashAgg(node) => Some(format!(
+                "state tables: [{}]",
+                node.internal_tables
+                    .iter()
+                    .map(|tb| { self.add_table(tb) })
+                    .join(", ")
+            )),
+            stream_node::NodeBody::AppendOnlyTopN(node) => Some(format!(
+                "state table: {}",
+                self.add_table(node.get_table().unwrap())
+            )),
+            stream_node::NodeBody::HashJoin(node) => Some(format!(
+                "left table: {}, right table {},{}{}",
+                self.add_table(node.get_left_table().unwrap()),
+                self.add_table(node.get_right_table().unwrap()),
+                match &node.left_degree_table {
+                    Some(tb) => format!(" left degree table: {},", self.add_table(tb)),
+                    None => "".to_string(),
+                },
+                match &node.right_degree_table {
+                    Some(tb) => format!(" right degree table: {},", self.add_table(tb)),
+                    None => "".to_string(),
+                },
+            )),
+            stream_node::NodeBody::TopN(node) => Some(format!(
+                "state table: {}",
+                self.add_table(node.get_table().unwrap())
+            )),
+            stream_node::NodeBody::Lookup(node) => Some(format!(
+                "arrange table: {}",
+                self.add_table(node.get_arrangement_table().unwrap())
+            )),
+            stream_node::NodeBody::Arrange(node) => Some(format!(
+                "arrange table: {}",
+                self.add_table(node.get_table().unwrap())
+            )),
+            stream_node::NodeBody::DynamicFilter(node) => Some(format!(
+                "left table: {}, right table {}",
+                self.add_table(node.get_left_table().unwrap()),
+                self.add_table(node.get_right_table().unwrap()),
+            )),
+            stream_node::NodeBody::GroupTopN(node) => Some(format!(
+                "state table: {}",
+                self.add_table(node.get_table().unwrap())
+            )),
+            _ => None,
+        };
+        if let Some(explain_table_oneline) = explain_table_oneline {
+            writeln!(f, "{}{}", " ".repeat(level * 2 + 1), explain_table_oneline)?;
+        }
+
+        if self.verbose {
```

Comment:
Could you please give an example what will it look like?

Reply:
yes, I add it in the PR description